### ISSUE TITLE
Add closing codes and reasons

### ISSUE THREAD

**Issue by Cyberfly100** [MediaMongrels-Ltd/LabVIEW-WebSockets-API#5]
The webocket protocol states that a reason can be given by the endpoint, when sending a closing frame. It should be a two byte code (1000 for "everything is fine") followed by an optional utf-8 encoded message. Python's websockets server generates an error, when no code is sent, assuming something went wrong.
It would be good to send a confirmation that closing was expected.
I could fix my issue by changing the "close frame.vi" as follows:
![image](https://user-images.githubusercontent.com/28017534/69961946-3e2b3a80-150d-11ea-8631-393bbfd0161d.png)
[Download (valid until 2020-01-31)](https://kth.box.com/s/9c1td2clp7fxtkuycct2skohei13smnw)

I would recommend to make the Ring Menu into a Control and use it as an optional input. Then one can program a client that automatically closes the connection with the correct error code, when e.g. a binary payload instead of an expected text payload is received.

**Comment by samsharp99** [comment]
Hi Lucas,

The [WebSockets specification](https://tools.ietf.org/html/rfc6455#section-5.5.1) states:

> The Close frame MAY contain a body (the "Application data" portion of
   the frame) that indicates a reason for closing

The closing body and status code is not _required_ by the WebSockets specification so if the Python library is returning an error then that is an issue with the Python library.

**Comment by Cyberfly100** [comment]
True. I also wrote that a reason CAN be given.
Not sure how well it is handled on the python side. The error is printed but the server keeps running happily. Maybe it is intended as a strong suggestion...
Anyways, I thought it is an easy thing to add and it could have sensibel default values (maybe empty by default, but "OK/end of use" when used in the example client at program stop). And it allows the server to know what went wrong in case something is not OK.

**Comment by samsharp99** [comment]
Yes, that's OK - I will leave the issue open and have no issue with implementing it at some stage but if the Python side is generating an exception/error then that's not an issue with the LabVIEW library.
